lib/matplotlib/_mathtext.py

```diff
@@ -66,6 +66,83 @@ def get_unicode_index(symbol, math=False):  # Publicly exported.
             .format(symbol)) from err
 
 
+VectorParse = namedtuple("VectorParse", "width height depth glyphs rects",
+                         module="matplotlib.mathtext")
+VectorParse.__doc__ = r"""
+The namedtuple type returned by ``MathTextParser("path").parse(...)``.
+
+This tuple contains the global metrics (*width*, *height*, *depth*), a list of
+*glyphs* (including their positions) and of *rect*\angles.
+"""
+
+
+RasterParse = namedtuple("RasterParse", "ox oy width height depth image",
+                         module="matplotlib.mathtext")
+RasterParse.__doc__ = r"""
+The namedtuple type returned by ``MathTextParser("agg").parse(...)``.
+
+This tuple contains the global metrics (*width*, *height*, *depth*), and a
+raster *image*.  The offsets *ox*, *oy* are always zero.
+"""
+
+
+class Output:
+    r"""
+    Result of `ship`\ping a box: lists of positioned glyphs and rectangles.
+
+    This class is not exposed to end users, but converted to a `VectorParse` or
+    a `RasterParse` by `.MathTextParser.parse`.
+    """
+
+    def __init__(self, box):
+        self.box = box
+        self.glyphs = []  # (ox, oy, info)
+        self.rects = []  # (x1, y1, x2, y2)
+
+    def to_vector(self):
+        w, h, d = map(
+            np.ceil, [self.box.width, self.box.height, self.box.depth])
+        gs = [(info.font, info.fontsize, info.num, ox, h - oy + info.offset)
+              for ox, oy, info in self.glyphs]
+        rs = [(x1, h - y2, x2 - x1, y2 - y1)
+              for x1, y1, x2, y2 in self.rects]
+        return VectorParse(w, h + d, d, gs, rs)
+
+    def to_raster(self):
+        xmin = min([*[ox + info.metrics.xmin for ox, oy, info in self.glyphs],
+                    *[x1 for x1, y1, x2, y2 in self.rects], 0]) - 1
+        ymin = min([*[oy - info.metrics.ymax for ox, oy, info in self.glyphs],
+                    *[y1 for x1, y1, x2, y2 in self.rects], 0]) - 1
+        xmax = max([*[ox + info.metrics.xmax for ox, oy, info in self.glyphs],
+                    *[x2 for x1, y1, x2, y2 in self.rects], 0]) + 1
+        ymax = max([*[oy - info.metrics.ymin for ox, oy, info in self.glyphs],
+                    *[y2 for x1, y1, x2, y2 in self.rects], 0]) + 1
```

Comment:
Yes, this is because half of the functions have y's increasing upwards and the other half is increasing downwards :/  I can try to rationalize or at least better document this another time, but I'll claim an out on this one (as a "it was like that before") for now.

Reply:
A comment that that is intended would still be good.

Reply:
sure, done.